### PR TITLE
add OffsetClock and add withZone and equals to other Clock impls

### PR DESCRIPTION
**Pull request by henryw374**

### CODE REVIEW

**Comment by henryw374** [comment]
cheers. I guess I need to bump up the coverage first anyway.... I'm assuming coveralls only accepts increase or same coverage

File: src/Clock.js [hunk 1]
```diff
@@ -139,6 +162,20 @@ export class Clock {
...
+     * Returns a copy of this clock with a different time-zone.
+     * <p>
+     * A clock will typically obtain the current instant and then convert that
+     * to a date or time using a time-zone. This method returns a clock with
+     * similar properties but using a different time-zone.
+     *
+     * @param zone  the time-zone to change to, not null
+     * @return a clock based on this clock with the specified time-zone, not null
+     */
+    withZone(){
```
[the remark this answers is not in the document]
Comment (henryw374) [reply src/Clock.js]: yes we need it :) . Long-ish I am maintaining a cross-platform clojure library that uses java.time and js-joda underneath, so any discrepancies between the two are potenitally a problem. one way we use clock is to fix the ambient time using out with-clock macro : https://github.com/henryw374/tick/blob/master/src/tick/alpha/api.cljc#L277 so if we wanted to simulate a user/vm in a particular zone then we'd use withZone

File: src/Clock.js [hunk 2]
```diff
@@ -104,6 +104,29 @@ export class Clock {
     static fixed(fixedInstant, zoneOffset) {
```
[the remark this answers is not in the document]
Comment (henryw374) [reply src/Clock.js]: ok, I'll change it

**Comment by henryw374** [comment]
Hi, I'll try to get around to adding the extra test. it would use the zone impls from CurrentStandardZone I guess.

Cheers.

File: src/Clock.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (pithu) [reply src/Clock.js]: ok

**Comment by pithu** [comment]
Yes `CurrentStandardZone` is the right choice.

**Pull request merged**